Fix plot function

jona-sassenhagen opened the pull request:
Adress comment wrt. ica.py

dengemann commented on mne/preprocessing/ica.py:
```diff
@@ -2180,14 +2180,18 @@ def _plot_corrmap(data, subjs, indices, ch_type, ica, label, show, outlines,
         figs = [_plot_corrmap(data[k:k + p], subjs[k:k + p],
-                indices[k:k + p], ch_type, ica, label)
+                indices[k:k + p], ch_type, ica, label,
+                outlines, cmap, contours, layout, show)
```
for avoding bugs we should pass the args by keywords. I'm sometimes too lazy to follow this advice but in general we should do this.

jona-sassenhagen commented:
@dengemann Is anything missing?

agramfort commented:
looks like @dengemann starts to need a clone...

Commit:
wip

teonbrooks committed:
ENH prevent break backwards compat

teonbrooks committed:
Minor fixes

teonbrooks committed:
Update tests with picks

teonbrooks committed:
FIX test

teonbrooks committed:
PEP8

teonbrooks committed:
ADD apply_baseline method

jona-sassenhagen committed:
if ch_type == None: ch_type = _get_ch_type(ica, ch_type)

wronk committed:
Initial tSSS implementation

wronk committed:
Update params, add tests

wronk committed:
FIX: bug in tSSS projector

dengemann committed:
simplify minimum norm TFR code

dengemann committed:
cosmit

dengemann committed:
fixe returns and unpacking

dengemann committed:
fix c+p error

dengemann committed:
add doc to manual

dengemann committed:
somits

jona-sassenhagen committed:
fix w/o from future import

agramfort committed:
Merge pull request #2455 from jona-sassenhagen/patch-13

Fix division bug

dengemann committed:
Merge pull request #2453 from dengemann/clean_up_mne_tfr

ENH: simplify minimum norm TFR code and imrpove readability

agramfort committed:
Merge pull request #2454 from christianbrodbeck/topomap

Minor FIX for topomap outlines:  set clip_on=False

agramfort committed:
Merge pull request #2401 from kingjr/GAT_chunk_scorer_parallel

EHN: GAT parallel across chunks for predict & score

dengemann committed:
more cleanups + better abstractions + option to return mere TFR

dengemann committed:
improve naming

larsoner committed:
FIX: Fix types

agramfort committed:
Merge pull request #2452 from Eric89GXL/consistency

MRG: Check info consistency

dengemann committed:
improve chunking and abstraction

larsoner committed:
Merge pull request #2456 from dengemann/clean_up_mne_tfr

ENH: improve structure of MNE-TFR code and isolate the single trial TFR computational step

dengemann committed:
make sure info is an info, plotting will break otherwise after reading

dengemann committed:
inplace operator power

larsoner committed:
Merge pull request #2458 from dengemann/fix_tfr_info_io

FIX: make sure info is an info, plotting will break otherwise after reading

larsoner committed:
Merge pull request #2459 from dengemann/improve_power_speed

ENH: inplace operator power

teonbrooks committed:
ENH helper function

larsoner committed:
FIX: Add some logging

Commit:
Improve extended infomax code such that mne_python and eeglab

teonbrooks committed:
ENH depr plot_image_epochs

teonbrooks committed:
ENH deprecate plot_image_epochs for plot_epochs_image

Commit:
Edit whats_new.rst file, and fix some revision comments.

agramfort committed:
Merge pull request #2460 from jmontoyam/EEGLAB_extended_infomax_comparison

Improve extended infomax code such that mne_python and eeglab obtain the same result when using the same parameters.

agramfort committed:
misc

agramfort committed:
FIX : make DICS work with volume source space

dengemann committed:
Merge pull request #2463 from agramfort/fix_dics_volume_source_space

[MRG] FIX : make DICS work with volume source space

teonbrooks committed:
Update whats_new

Remove trailing whitespace
Add function handle to mne.Epochs.plot
Add new function apply_baseline

teonbrooks committed:
PEP8 [ci skip]

larsoner committed:
Merge pull request #2337 from teonlamont/epochs

[MRG] ENH add baseline arg to read_epochs

teonbrooks committed:
Update whats_new

teonbrooks committed:
Update whats_new

larsoner committed:
ENH: Allow preload=False in read_epochs

larsoner committed:
Merge pull request #2461 from teonlamont/depr

[MRG] ENH depr plot_image_epochs

wronk committed:
ENH: Better handling of buffer windows, and clarifications in var names

larsoner committed:
ENH: Slightly better dipole accuracy

Commit:
Address comments

larsoner committed:
FIX: File descriptors and unnecessary read

agramfort committed:
Merge pull request #2468 from Eric89GXL/fits

MRG: Slightly better dipole accuracy

larsoner committed:
Merge pull request #2309 from teonlamont/tf

[MRG] ENH add picks argument to tfr_morlet, tfr_multitaper

larsoner committed:
FIX: Code literal

larsoner committed:
FIX: Update whats_new.py

agramfort committed:
Merge pull request #2466 from Eric89GXL/epochs-read

MRG: Allow preload=False in read_epochs

larsoner committed:
FIX: Unify BTI transforms

larsoner committed:
FIX: Fix for n_jobs source space

agramfort committed:
Merge pull request #2474 from Eric89GXL/dist-fix

MRG: Fix for n_jobs source space

larsoner committed:
FIX: Better use of existing functions

dengemann committed:
test io

dengemann committed:
io support for dev_ctf_t

wmvanvliet committed:
Add more error messages to viz.plot_topomap

The `pos` parameter of the `viz.plot_topomap` function expects a matrix
with shape (#nchannels, 2). It used to silently accept other dimensions
as well by selecting `pos[:, 2]`. This means 3D coordinates, such as the
ones provided by Montage objects are accepted and lead to strange
results. See issue #2389.

This PR adds error messages when a matrix with invalid dimensions is
specified as `pos` argument. To keep backwards compatibility, matrices
with shape (#nchannels, 4) are allowed, meaning a user can still pass a
`layout.pos` matrix, even though `layout.pos[:, 2]` would be more
explicit and thus preferred.

dengemann committed:
add fix + 1 more test

dengemann committed:
get trans from fiducials

larsoner committed:
Merge pull request #21 from dengemann/refactor-trans

ENH: Refactor trans

larsoner committed:
FIX: Minor fixes

larsoner committed:
FIX: Fix KIT

dengemann committed:
Merge pull request #2473 from Eric89GXL/refactor-trans

MRG: Unify BTI transforms

larsoner committed:
ENH: Verbose output for plot_trans

wmvanvliet committed:
matrix -> 2D array

wmvanvliet committed:
Merge pull request #2475 from wmvanvliet/topomap-error

Add some error messages to `viz.plot_topomap`

larsoner committed:
ENH: Refactor transforms

larsoner committed:
FIX: Fix colors [ci skip]

dengemann committed:
Merge pull request #2477 from Eric89GXL/plot-tweaks

MRG: Verbose output for plot_trans

larsoner committed:
FIX: Robust fitting for sphere

agramfort committed:
Merge pull request #2479 from Eric89GXL/robust-fit

MRG: Robust fitting for sphere

dengemann committed:
add first test on transforms

dengemann committed:
de-convolute bti headshape io

dengemann committed:
more de-convolution

dengemann committed:
add no-convert option

dengemann committed:
satisy @Eric89GXL

dengemann committed:
one more

dengemann committed:
Merge pull request #2480 from dengemann/bti-tweaks

ENH: more bti tweaks

agramfort committed:
iter on simulation code

larsoner committed:
WIP: More complete simulation

larsoner committed:
FIX: Merge fixes

larsoner committed:
FIX: Many fixes

larsoner committed:
FIX: Remove print

kingjr committed:
FIX: when n_jobs > n_test_times, remove empty jobs

larsoner committed:
FIX: Many fixes

larsoner committed:
FIX: Fix example

larsoner committed:
FIX: Fix test

larsoner committed:
FIX: Fix tests more

larsoner committed:
FIX: Fix for py26

wronk committed:
Fix factor typo

agramfort committed:
Merge pull request #2484 from wronk/stats_typo

[MRG] Fix factor typo

kingjr committed:
fix n_chunk

agramfort committed:
Merge pull request #2483 from kingjr/fix_gat_njobs

FIX: when n_jobs > n_test_times in GAT, remove empty jobs

larsoner committed:
FIX: Better coverage

agramfort committed:
FIX : ica.apply(raw) with ref_meg channels

larsoner committed:
Merge pull request #2485 from agramfort/fix_ref_meg_ica

FIX : ica.apply(raw) with ref_meg channels

larsoner committed:
FIX: Address first round of comments

dengemann committed:
base channel iteration on 4D labels

dengemann committed:
add test for ch renaming

larsoner committed:
FIX: Multiple fixes

dengemann committed:
make reoridering optional

dengemann committed:
consistency

larsoner committed:
FIX: Make inv["noise_cov"] a Covariance

dengemann committed:
bytes-io-api

agramfort committed:
API : add option not to preload ECG epochs (can be big)

agramfort committed:
do the same for EOG

dengemann committed:
bytes-io-test+workaround-np.fromfile

agramfort committed:
Merge pull request #2481 from Eric89GXL/simulation_eval

MRG: Simulation eval

dengemann committed:
Merge pull request #2488 from Eric89GXL/cov

MRG: Make inv["noise_cov"] a Covariance

agramfort committed:
update what's new

agramfort committed:
clarify docstrings of XDAWN wrt overfit

agramfort committed:
typo

agramfort committed:
API : rename compute_raw_data_covariance to compute_raw_covariance

dengemann committed:
Merge pull request #2492 from agramfort/no_preload_ssp

API : add option not to preload ECG epochs (can be big)

larsoner committed:
FIX: Fix test

dengemann committed:
Merge pull request #2495 from agramfort/rename_compute_raw_data_covariance

API : rename compute_raw_data_covariance to compute_raw_covariance

dengemann committed:
numpy instead of struct + private functions

dengemann committed:
py3k fixes

dengemann committed:
np.fromstring in other places except externals and appveyor.yml @Eric89GXL

dengemann committed:
[ci-skip] revert example

dengemann committed:
be precise about types

larsoner committed:
Merge pull request #2487 from dengemann/bti-tweaks

ENH: more bti tweaks

larsoner committed:
FIX: Fix type

larsoner committed:
FIX: Move files

larsoner committed:
Merge pull request #2496 from Eric89GXL/minor-fix

FIX: Fix type

wronk committed:
Fix: last window handling. Add to docs

wronk committed:
Update tests

teonbrooks committed:
MV preload_data load_data

larsoner committed:
FIX: Localization working

larsoner committed:
FIX: Fix init

wronk committed:
FIX: Rebase cleanup

larsoner committed:
FIX: Better formatting

larsoner committed:
FIX: Fix for bad finding

teonbrooks committed:
Minor fix

larsoner committed:
FIX: Better coverage

dengemann committed:
Merge pull request #2498 from teonlamont/load_data

Rename `preload_data` method to `load_data`

teonbrooks committed:
Depr preload_data

larsoner committed:
FIX: More efficient

larsoner committed:
FIX: Tweak tests

larsoner committed:
FIX: Fix for g

teonbrooks committed:
Update whats_new

teonbrooks committed:
FIX

larsoner committed:
ENH: Lower memory usage

larsoner committed:
ENH: Better docstring tests

larsoner committed:
FIX: Fix for old scipy

larsoner committed:
ENH: Enable fixing of mag channels

larsoner committed:
FIX: Restore tests

larsoner committed:
FIX: Fix comment [ci skip]

dengemann committed:
partially vectorize components2list

agramfort committed:
Merge pull request #2503 from Eric89GXL/fix-type

MRG: Enable fixing of mag channels

agramfort committed:
Merge pull request #2499 from Eric89GXL/chpi

MRG: cHPI position estimation

dengemann committed:
tiny edits

wronk committed:
Add/reconfigure tests, small doc update

dengemann committed:
Merge pull request #2504 from dengemann/faster_clustering_permutations

ENH: partially vectorize components2list

larsoner committed:
Merge pull request #2400 from wronk/tSSS

[MRG] SSS: tSSS

larsoner committed:
FIX: Update for SSS

jona-sassenhagen committed:
Automatically detect peaks

jona-sassenhagen committed:
Update topomap.py

jona-sassenhagen committed:
Update topomap.py

jona-sassenhagen committed:
Update topomap.py

jona-sassenhagen committed:
Update topomap.py

jona-sassenhagen committed:
max 10 topos, fix nested import

jona-sassenhagen committed:
test for peaks functionality

jona-sassenhagen committed:
Update evoked.py

jona-sassenhagen committed:
pep 8

jona-sassenhagen committed:
Update test_topomap.py

jona-sassenhagen committed:
Dealing with cases where no peak is detected

jona-sassenhagen committed:
pep8

Commit:
FIX verbose

dengemann committed:
more doc

dengemann committed:
cosmits

agramfort committed:
Merge pull request #2508 from dengemann/doc_bem

DOC: more doc

Commit:
squash

teonbrooks committed:
ADD add'l bef setting

dengemann committed:
stricter coord frames

dengemann committed:
more edits on bti

jaeilepp committed:
Fix to ica.plot_sources.

jona-sassenhagen committed:
spelling

jona-sassenhagen committed:
Update test_utils.py

agramfort committed:
Merge pull request #2513 from jaeilepp/dont-show-refmeg-in-ica

Fix to ica.plot_sources.

jaeilepp committed:
Topomaps by selecting an area in evoked.plot.

jaeilepp committed:
Cleaning

jona-sassenhagen committed:
Update ica.py

Commit:
improve example a bit

teonbrooks committed:
Merge pull request #2510 from teonlamont/bef

[MRG] ADD add'l bef setting

jaeilepp committed:
Changed the topomap to show average.

jaeilepp committed:
Added test.

jaeilepp committed:
Added channel names to titles.

jaeilepp committed:
Bad channel names gray.

jaeilepp committed:
Added note to example.

larsoner committed:
ENH: Tweak simulation

larsoner committed:
Merge pull request #2515 from jaeilepp/channel-names-gray-on-raw-plot

[MRG] Bad channel names gray.

larsoner committed:
Merge pull request #2501 from teonlamont/depr

[MRG] Depr preload_data

jaeilepp committed:
Fix to whats_new.

larsoner committed:
FIX: Unify BrainVision reading

larsoner committed:
FIX: Fix for old numpy

jaeilepp committed:
Disabled blit.

larsoner committed:
Merge pull request #2516 from Eric89GXL/sim-tweaks

MRG: Tweak simulation

jaeilepp committed:
Backend queried for blit.

larsoner committed:
FIX: Address comments

larsoner committed:
Merge pull request #2517 from Eric89GXL/brainvision

FIX: Unify BrainVision reading

jaeilepp committed:
All channel types to one figure.

jaeilepp committed:
Fix to test.

jaeilepp committed:
Added feedback.

jaeilepp committed:
Custom size to remove empty space.

jaeilepp committed:
Indicators of selected area. Title to bottom.

larsoner committed:
Merge pull request #2514 from jaeilepp/topomap-from-evoked-plot

Topomap from evoked plot

larsoner committed:
FIX: Fix message

larsoner committed:
ENH: Add source space morphing

jona-sassenhagen committed:
fix completely dry runs in case (no dict creation)

jaeilepp committed:
Fix to tests.

jaeilepp committed:
Ten biggest peaks selected.

jaeilepp committed:
Fix.

larsoner committed:
FIX: Fix division

larsoner committed:
FIX: Fix docstring

larsoner committed:
FIX: Fix Travis

larsoner committed:
Merge pull request #2527 from Eric89GXL/minor-fix

FIX: Fix division

larsoner committed:
FIX: Factor out eeg_loc and coil_trans

larsoner committed:
FIX: Fix test

larsoner committed:
FIX: Better comment

jaeilepp committed:
Peak finder as private function. Tests.

jaeilepp committed:
fix

larsoner committed:
FIX: Minor fixes

larsoner committed:
Merge pull request #2525 from jaeilepp/autoselect-ch_type

[MRG] Autoselect ch type

larsoner committed:
Merge pull request #2512 from dengemann/bti-tweaks

WIP/ENH: more bti tweaks

larsoner committed:
FIX: Minor fixes

agramfort committed:
Merge pull request #2528 from Eric89GXL/no-ct

MRG: Factor out eeg_loc and coil_trans

jaeilepp committed:
Fix to accept axes.

jaeilepp committed:
Updated whats_new.

jaeilepp committed:
Merge pull request #2526 from jaeilepp/peak-find

[MRG] Automatic peak finder

larsoner committed:
FIX: Fix indexing

larsoner committed:
ENH: Unify src use

larsoner committed:
ENH: Add compare_fiff

larsoner committed:
ENH: Chaining

larsoner committed:
FIX: Fix usage

agramfort committed:
Merge pull request #2529 from Eric89GXL/compare

MRG: Add compare_fiff

larsoner committed:
FIX: Travis workaround

larsoner committed:
FIX: Update whats_new

larsoner committed:
Merge pull request #2522 from Eric89GXL/morph

MRG: Add source space morphing

agramfort committed:
Merge pull request #2531 from Eric89GXL/trav-fix

FIX: Travis workaround

larsoner committed:
FIX: Minor fixes

jona-sassenhagen committed:
minor style thing
